### ISSUE TITLE
Support for Truffle plugins in Yarn workspaces

### ISSUE THREAD

**Issue by rkalis** [trufflesuite/truffle#2409]
## Issue

@PaulRBerg raised an [issue in truffle-plugin-verify](https://github.com/rkalis/truffle-plugin-verify/issues/10) about the plugin not working in Yarn workspaces. The contents of his issue:

---

Yarn workspaces is this new way of intelligently managing your npm modules in a monorepo by installing them only once (or multiple times if there are overlapping versions). What it does is that it aggregates all the modules at the root of your project.

If I install truffle-plugin-verify in the "contracts" package of this monorepo and I try to run it, I get the following error:

> Error: truffle-plugin-verify listed as a plugin, but not found in global or local node modules!

Of course, I moved on and installed the module globally, but I can't use this plugin on CI.

---

I'm guessing that Truffle searches for installed plugins in the `node_modules` of the directory containing `truffle.js`, while in this case the plugin is installed in the root `node_modules` of the workspace, several levels higher.

## Steps to Reproduce

1. Start a project with Yarn workspaces.
2. Start a Truffle project inside this project.
3. Install a Truffle plugin (e.g. `truffle-plugin-verify`) into the Truffle project.
4. Try to use the plugin (e.g. `truffle plugin run verify Roscoin --network goerli`).

## Expected Behavior

Acknowledge that the plugin is installed into the root of the project rather than the current directory, and access it from there.

## Actual Results

> Error: truffle-plugin-verify listed as a plugin, but not found in global or local node modules!

## Environment

* Operating System: ?
* Ethereum client: N/A
* Truffle version (`truffle version`): 5.0.?
* node version (`node --version`): ?
* npm version (`npm --version`): ?

**Comment by PaulRBerg** [comment]
Thanks @rkalis!

## Environment

- Operating System: macOS 10.14.6
- Truffle Version: 5.0.35
- node version: 10.15.3
- npm: 6.11.3

**Comment by gnidan** [comment]
Thanks both of you! We'll triage this and get it sorted out.

**Comment by sirhill** [comment]
Creating a symlink from the parent node_modules to the repo node_modules seems to be a valid workaround.

**Comment by PaulRBerg** [comment]
Yeah, that works. I learned about the [nohoist](https://classic.yarnpkg.com/blog/2018/02/15/nohoist/) option in the meantime:

```js
{
    "workspaces": {
      "packages": [
        "packages/*"
      ],
      "nohoist": [
        "**/package-name-here",
        "**/package-name-here/**"
      ]
  }
}
```

**Comment by 6gunner** [comment]
Is there any way to specify my plugin directory through truffle.config.js?   Usually I just want to write a special plugin for my project, and I don't wanna publish or install it thorugh npm

**Comment by rkalis** [comment]
Iirc you can also specify relative paths for specific plugins if it's a local plugin. Note that you do need to add a "truffle-plugin.json" file in the directory of your local plugin. Perhaps @gnaths can confirm?
